pkg/expr/graph.go

```diff
@@ -144,7 +149,7 @@ func (s *Service) buildGraph(req *backend.QueryDataRequest) (*simple.DirectedGra
 		case dsName == DatasourceName || dsUID == DatasourceUID:
 			node, err = buildCMDNode(dp, rn)
 		default: // If it's not an expression query, it's a data source query.
-			node, err = s.buildDSNode(dp, rn, req.PluginContext.OrgID)
+			node, err = s.buildDSNode(dp, rn, req.OrgId)
```

Comment:
Not sure I follow. But further down (using the bus, should probably fix this), the data sources will be queried based on the OrgID+UID, so will only get plugins if they are in the org - I think.

But in terms of permissions, currently that needs to be checked before expressions, since it isn't passing a signed in user.

Reply:
> Not sure I follow. But further down (using the bus, should probably fix this), the data sources will be queried based on the OrgID+UID, so will only get plugins if they are in the org - I think.

👍  so any request for a datasource belonging to a different organisation will eventually fail.